core/templates/pages/exploration-editor-page/services/state-tutorial-first-time.service.ts

```diff
@@ -16,100 +16,111 @@
-          _openPostTutorialHelpPopoverEventEmitter.emit();
-          EditorFirstTimeEventsService.registerEditorFirstEntryEvent();
-        }
-
-        _currentlyInEditorFirstVisit = false;
-      },
-      markTranslationTutorialNotSeenBefore: function() {
-        _translationTutorialNotSeenBefore = true;
-      },
-      initTranslation: function(expId) {
-        // After the first call to it in a client session, this does nothing.
-        if (!_translationTutorialNotSeenBefore ||
-            !_currentlyInTranslationFirstVisit) {
-          _currentlyInTranslationFirstVisit = false;
-        }
-
-        if (_currentlyInTranslationFirstVisit) {
-          enterTranslationForTheFirstTimeEventEmitter.emit();
-          EditorFirstTimeEventsService.initRegisterEvents(expId);
-          $http.post(STARTED_TRANSLATION_TUTORIAL_EVENT_URL + '/' + expId)
-            .then(null, function() {
-              console.error(
-                'Warning: could not record translation tutorial start event.'
-              );
-            });
-        }
-      },
-      markTranslationTutorialFinished: function() {
-        if (_currentlyInTranslationFirstVisit) {
-          _openPostTutorialHelpPopoverEventEmitter.emit();
-          EditorFirstTimeEventsService.registerEditorFirstEntryEvent();
-        }
-
-        _currentlyInTranslationFirstVisit = false;
-      },
-      get onEnterEditorForTheFirstTime() {
-        return enterEditorForTheFirstTimeEventEmitter;
-      },
-      get onEnterTranslationForTheFirstTime() {
-        return enterTranslationForTheFirstTimeEventEmitter;
-      },
-      get onOpenEditorTutorial() {
-        return _openEditorTutorialEventEmitter;
-      },
-      get onOpenPostTutorialHelpPopover() {
-        return _openPostTutorialHelpPopoverEventEmitter;
-      },
-      get onOpenTranslationTutorial() {
-        return _openTranslationTutorialEventEmitter;
-      }
-    };
+import { EventEmitter, Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { EditorFirstTimeEventsService } from 'pages/exploration-editor-page/services/editor-first-time-events.service.ts';
+import { TutorialEventsBackendApiService } from 'pages/exploration-editor-page/services/tutorial-events-backend-api.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StateTutorialFirstTimeService {
+  // Whether this is the first time the tutorial has been seen by this user.
+  _currentlyInEditorFirstVisit: boolean = true;
+  _currentlyInTranslationFirstVisit: boolean = true;
+  _translationTutorialNotSeenBefore: boolean = false;
+
+  private enterEditorForTheFirstTimeEventEmitter = new EventEmitter();
+  private enterTranslationForTheFirstTimeEventEmitter = new EventEmitter();
+
+  _openEditorTutorialEventEmitter = new EventEmitter();
+  _openPostTutorialHelpPopoverEventEmitter = new EventEmitter();
+  _openTranslationTutorialEventEmitter = new EventEmitter();
```

Comment:
are we expecting these variables to be private?

Reply:
The two variables were annotated with /** @private */ before migrating.

Reply:
Should we use use private here [and for all the service properties starting with `_`]?

(Earlier the code where not in es6, so we weren't using private modifiers.)

Reply:
Done